src/main/java/org/broadinstitute/hellbender/utils/config/ConfigUtils.java

```diff
@@ -0,0 +1,332 @@
+     * @param argList The list of arguments from which to read the config file.
+     * @param configFileOption The command-line option specifying the main configuration file.
+     * @param configClazz The class of the configuration file to instantiate.
+     */
+    public static final <T extends Config> void initializeConfigurationsFromCommandLineArgs(final ArrayList<String> argList,
+                                                                                            String configFileOption,
+                                                                                            Class<? extends T> configClazz) {
+        Utils.nonNull(argList);
+        Utils.nonNull(configFileOption);
+
+        // Get main config from args:
+        final String configFileName = getConfigFilenameFromArgs( argList, configFileOption );
+
+        // Set the config file to be the one we want to use from the command-line:
+        ConfigFactory.setProperty( GATKConfig.CONFIG_FILE_VARIABLE_NAME, configFileName );
+        T gatkConfig = ConfigCache.getOrCreate(configClazz);
+
+//        // Alternate way to load the config file:
+//        T gatkConfig = ConfigUtils.initializeConfigurationAsProperties( configFileName, configClazz );
+
+        // To start with we inject our system properties to ensure they are defined for downstream components:
+        ConfigUtils.injectSystemPropertiesFromConfig( gatkConfig, GATKConfig.SYSTEM_PROPERTY_PREFIX );
+    }
+
+    /**
+     * Initializes and returns the configuration as specified by {@code configFileName}
+     * Also caches this configuration in the {@link ConfigCache} for use elsewhere.
+     * @param configFileName The name of the file from which to initialize the configuration
+     * @param configClass The type of configuration in which to interpret the given {@code configFileName}
+     * @return The configuration instance implementing {@link GATKConfig} containing any overrides in the given file.
+     */
+    public static final <T extends Config> T initializeConfigurationAsProperties(final String configFileName, Class<? extends T> configClass) {
+
+        Utils.nonNull(configClass);
+
+        // Get a place to store our properties:
+        final Properties userConfigFileProperties = new Properties();
+
+        // Try to get the config from the specified file:
+        if ( configFileName != null ) {
+
+            try {
+                final FileInputStream userConfigFileInputStream = new FileInputStream(configFileName);
+                userConfigFileProperties.load(userConfigFileInputStream);
+
+                if (configFileName != null) {
+                    System.out.println("Found " + configClass.getSimpleName() + " Configuration File: " + configFileName);
+                }
+
+            } catch (final FileNotFoundException e) {
+                System.err.println("WARNING: unable to find specified " + configClass.getSimpleName() + " configuration file: "
```

Comment:
I think we might want to die here if you can't load the specified config file instead of continuing with a warning.